### ISSUE TITLE
NotOptimalIfConditionsInspection warning for 2 boolean expression and 4 cases

### ISSUE THREAD

**Issue by puggan** [kalessil/phpinspectionsea#1701]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Ultimate) v3.0.10 |
| Language level | PHP 7.0 |

Exemple code:
```php
(static function () {
    $a = random_int(0, 100);
    $b = random_int(0, 100);

    if ($a > 50) {
        /** @noinspection NotOptimalIfConditionsInspection */
        if ($b > 50) {
            echo 'AB';
        } else {
            echo 'Ab';
        }
    } elseif ($b > 50) {
        echo 'aB';
    } else {
        echo 'ab';
    }
})();
```

Question
--
Is this a false-positive case, or how am I to improve the if above?
from what I can read out of it, the `if ($b > 50)` and the `elseif ($b > 50)` is on 2 diffrenet branches, and is only one of them is going to be executed.

Or is a bitmask switch preferable in this case?
```php
switch ( ($a > 50 ? 0x1 : 0) + ($b > 50 ? 0x2 : 0) ) {
   case 0x0:
      echo 'ab';
      break;
   case 0x1:
      echo 'Ab';
      break;
   case 0x2:
      echo 'aB';
      break;
   case 0x3:
      echo 'AB';
      break;
}
```

**Comment by ea-inspections-team** [comment]
Thanks for reporting @puggan !
The issue should be addressed already, the detection of the duplicate conditions was dropped in July 2020.
